### ISSUE TITLE
Intermittent CI failures on `quic::test::test_quic_server_multiple_writes`

### ISSUE THREAD

**Issue by steviez** [solana-labs/solana#29096]
#### Problem
I'm seeing intermittent failures on the test `quic::test::test_quic_server_multiple_writes`. I'm not the only who has seen this, there was some [chatter about this in Discord](https://discord.com/channels/428295358100013066/560503042458517505/1040340055648116806) recently.

```
thread 'quic::test::test_quic_server_multiple_writes' panicked at 'called `Result::unwrap()`
on an `Err` value: Stopped(0)', streamer/src/nonblocking/quic.rs:1137:40
```
Here is a [build link](https://buildkite.com/solana-labs/solana/builds/85997#0184e67a-89c4-4d32-81f6-258106a15739) where I encountered this (twice thus far).

Brooks was previously unable to reproduce locally; I spun something up to see if I'm able to get any failures and will report back.

#### Proposed Solution
Debug and fix; Alessandro mentioned a theory about why the test became flaky in the flow that I linked above; after some discussion that theory was maybe disproved.

**Comment by brennanwatt** [comment]
+1 I've seen this failure as well. Was unable to reproduce locally on Macbook running 100+ serial iterations

**Comment by brennanwatt** [comment]
Note I'm still seeing this error intermittently.

I've been unable to reproduce locally on Macbook or Linux box when running hundreds of iterations in serial. Have attempted running just the single test:
`for i in {1..100}; do cargo test --package solana-streamer --lib -- quic::test::test_quic_server_multiple_writes --exact --nocapture; done`
Have also attempted running the quic suite:
`for i in {1..100}; do cargo test --package solana-streamer; done`

Digging into the failure signature, here are the major steps:
1. open QUIC connection
2. open outgoing unidirectional stream
3. write a packet's worth of data (1232B) one byte at a time to the stream
Step 3 is where we panic because we expect success but will occasionally receive `Stopped(0)` status indicating the peer is no longer accepting data on this stream. Note the 0 parameter is an application defined error code

So the question is why does the peer stop accepting data on the stream? I'm sure we could build in error handling on the client side, but I believe the intention of the test is to ensure writes succeed the first time

**Comment by behzadnouri** [comment]
I have seen this test failing repeatedly in `coverage` stage in CI; eg.
https://buildkite.com/solana-labs/solana/builds/87703

I have not seen the test fail in ci steps other than coverage.
Maybe getting ssh access to one of those ci machines and running the test with coverage flags would help to reproduce this.
Another possibility is that the test only fails with coverage flags _and_ while concurrently running with other tests (so not in isolation) perhaps due to higher concurrent load.

It might also be good to somehow ignore or bypass the test for coverage ci while this issue is being worked on, since it is really slowing things down.

**Comment by brennanwatt** [comment]
I am now able to reproduce this issue after creating the following unit test to essentially run hundreds of instances of `quic::test::test_quic_server_multiple_writes` in parallel:

```
    #[test]
    fn test_quic_server_multiple_writes_parallel() {
        let mut handles = vec![];
        for i in 0..1000 {
            handles.push(thread::spawn(move || {
                println!("spawning thread {}", i);
                test_quic_server_multiple_writes();
            }));
        }

        for handle in handles {
            handle.join().unwrap();
        }
    }
```

**Comment by brennanwatt** [comment]
I believe the root cause of this issue is the following:
1. We timeout waiting for chunks from the stream receiver on the server side
2. We drop the stream receiver
3. This triggers a Stopped frame to be sent
4. The client gets WriteError::Stopped status
5. Test fails

The chunk timeout gets set in `setup_quic_server` to 100ms, which is much lower than most places (set 1s or 10s in most places). I'm thinking we should make a constant for this value, increase to 1000ms, and stop seeing this failure.

**Comment by ilya-bobyr** [comment]
Interesting that it is set to `100ms` only in [`solana_streamer::quic::test::setup_quic_server`](https://github.com/solana-labs/solana/blob/ecea802fe66cb86d1c0be8a4c2de2fb4b1832686/streamer/src/quic.rs#L377), while in [`solana_streamer::nonblocking::quic::test`](https://github.com/solana-labs/solana/blob/d611cc2e75fc6b525a1b8b28428f1e4fc74fbc32/streamer/src/nonblocking/quic.rs#L1035) it is set to `1000ms`.
This probably explains why it is only the `solana_streamer::quic::test::setup_quic_server` test that is failing, while the `nonblocking` version, which is very similar otherwise, is not failing.

After reading the code for a bit, I think, increasing the timeout is the only reasonable choice.

I wonder if on a hyperthreaded CPU `--test-threads` which controls how many threads are executed is set to twice the actual number of cores.  And for test that have no IO, it causes noticeable slowdowns?
At the same time, 100ms is quite a lot.

**Comment by brennanwatt** [comment]
> Interesting that it is set to `100ms` only in [`solana_streamer::quic::test::setup_quic_server`](https://github.com/solana-labs/solana/blob/ecea802fe66cb86d1c0be8a4c2de2fb4b1832686/streamer/src/quic.rs#L377), while in [`solana_streamer::nonblocking::quic::test`](https://github.com/solana-labs/solana/blob/d611cc2e75fc6b525a1b8b28428f1e4fc74fbc32/streamer/src/nonblocking/quic.rs#L1035) it is set to `1000ms`. This probably explains why it is only the `solana_streamer::quic::test::setup_quic_server` test that is failing, while the `nonblocking` version, which is very similar otherwise, is not failing.
> 
> After reading the code for a bit, I think, increasing the timeout is the only reasonable choice.
> 
> I wonder if on a hyperthreaded CPU `--test-threads` which controls how many threads are executed is set to twice the actual number of cores. And for test that have no IO, it causes noticeable slowdowns? At the same time, 100ms is quite a lot.

Yes, exactly. There are a couple other tests with the lower, 100ms timeout, and I have a faint memory of seeing these tests fail sporadically (albeit at lower rate than `quic::test::test_quic_server_multiple_writes`) in the past as well
